include/sisl/fds/stream_tracker.hpp

```diff
@@ -88,6 +88,18 @@ class StreamTracker {
         m_comp_slot_bits.set_bits(start_bit, end_idx - start_idx + 1);
     }
 
+    void rollback(int64_t new_end_idx) {
+        folly::SharedMutexWritePriority::ReadHolder holder(m_lock);
```

Comment:
Should this be a Write lock since we're mutating the members below?

Reply:
The write lock is only for expansion and not changing individual bits. We still need to take ReadLock to prevent from someone else resizing them. We use m_comp_slot_bits and m_active_slot_bits are of type AtomicBitSet
https://github.com/hkadayam/sisl/blob/tracker_rollback/include/sisl/fds/stream_tracker.hpp#L329

AtomicBitSet explanation is given here
https://github.com/hkadayam/sisl/blob/tracker_rollback/include/sisl/fds/bitset.hpp#L1246